sdk/openai/openai/samples/cookbook/speechToSpeechChat/src/index.js

```diff
@@ -0,0 +1,181 @@
+const { OpenAIClient, AzureKeyCredential } = require("@azure/openai");
+
+let SpeechSDK, speechKey, speechRegion, recognizer, client, statusDiv, useAzureSpeechSDK;
+const azureSpeechCheckbox = document.getElementById('useAzureSpeechSDK');
+const speechKeyElement = document.getElementById("speechKey");
+const speechRegionElement = document.getElementById("speechRegion");
+azureSpeechCheckbox.addEventListener('change', function() {
+  useAzureSpeechSDK = this.checked;
+  speechKeyElement.disabled = !useAzureSpeechSDK;
+  speechRegionElement.disabled = !useAzureSpeechSDK;
+});
+
+function startChatFromSpeech() {
+  statusDiv = document.getElementById("statusDiv");
+  const promptInput = document.getElementById("promptInput");
+  statusDiv.innerHTML = "";
+  promptInput.innerHTML = "";
+
+  if (useAzureSpeechSDK) {
```

Comment:
Since we're using WebPack or some other bundler, we shouldn't be afraid to use multiple files, for example one that encapsulates the Web Speech API versus the Azure Speech SDK and import accordingly.

Reply:
That's fair, I'll look into splitting those functions and updating.